jscomp/main/js_main.ml

```diff
@@ -113,6 +113,97 @@ let usage = "Usage: bsc <options> <files>\nOptions are:"
 let ppf = Format.err_formatter
 
 (* Error messages to standard error formatter *)
+open struct
+  open Reason_migrate_parsetree
+  module To_current = Convert(OCaml_406)(OCaml_current)
+  module From_current = Convert(OCaml_current)(OCaml_406)
+
+  let handle_res_parse_result (parse_result : _ Res_driver.parseResult) =
+    if parse_result.invalid then begin
+        Res_diagnostics.printReport parse_result.diagnostics parse_result.source;
+        exit 1
+    end
+end
+
+let print_res_interface ~comments ast =
+  Res_printer.printInterface ~width:100 ~comments ast
+
+let print_res_implementation ~comments ast =
+  Res_printer.printImplementation ~width:100 ~comments ast
+
+(* TODO: support printing from AST too. *)
```

Comment:
Let's add an issue for this, as it is really useful especially if using ppx

Reply:
Agree. It can also be used for snapshot testing a ppx.

Reply:
#88